pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/LuceneFSTIndexReader.java

```diff
@@ -52,7 +52,8 @@ public LuceneFSTIndexReader(PinotDataBuffer pinotDataBuffer)
     _dataBufferIndexInput = new PinotBufferIndexInput(_dataBuffer, 0L, _dataBuffer.size());
 
     _readFST =
-        new FST(_dataBufferIndexInput, _dataBufferIndexInput, PositiveIntOutputs.getSingleton(), new OffHeapFSTStore());
+        new FST<>(FST.readMetadata(_dataBufferIndexInput, PositiveIntOutputs.getSingleton()),
```

Comment:
Introduced new static class [FSTMetadata](https://lucene.apache.org/core/9_9_0/core/org/apache/lucene/util/fst/FST.FSTMetadata.html) in lucene 9.9.0. This is getting used in FST constructor.

Reply:
what about the OffheapFSTStore?

In recent tests, I found the default became OnheapFSTStore.
```
  public FST(FSTMetadata<T> metadata, DataInput in) throws IOException {
    this(metadata, new OnHeapFSTStore(DEFAULT_MAX_BLOCK_BITS, in, metadata.numBytes));
  }
```

I'm looking for how to initialize it with OffheapFSTStore. Feel free to chime in.